sdk/Sdk/FunctionMetadataGenerator.cs

```diff
@@ -425,18 +459,25 @@ private static ExpandoObject BuildBindingMetadataFromAttribute(CustomAttribute a
                 bindingDict["Name"] = parameterName!;
             }
 
+            var direction = GetBindingDirection(attribute);
+            bindingDict["Direction"] = direction;
             bindingDict["Type"] = bindingType;
-            bindingDict["Direction"] = GetBindingDirection(attribute);
 
+            // For extensions that support worker binding, set the DataType so parameters are bound by the worker.
+            // Only use worker binding for input and trigger bindings
+            if (supportsReferenceType && direction != Constants.OutputBindingDirection)
```

Comment:
We probably don't need to have a check for output binding here because when we call this method (`BuildBindingMetadataFromAttribute`) from the output binding path, we always set supportsReferenceType to false

https://github.com/Azure/azure-functions-dotnet-worker/blob/08e86a23c00fd8199639c4588397b88c6bebdcae/sdk/Sdk/FunctionMetadataGenerator.cs#L444-L449